Feat/dev 161 remove duplicated code

asurinsaka opened the pull request:
refactor and remove duplicated code in tests/.

asurinsaka committed:
DEV-161 refactor tests

refactor code in test_query and confest

asurinsaka committed:
DEV-161 refactor mock_server fixture

move the fixture code of mock_server to conftest.py

anmaxvl commented on tests/conftest.py:
```diff
@@ -37,76 +50,44 @@ def make_tarfile(ids, tarfile_name="temp.tar", write_mode="w"):
+        "controlled", "small content 4", [], [],
+    ),
+    "big": generate_metadata_dict(
+        "controlled", next(big_content), ["annotation 1"], ["related 1"],
```
I think it'd make sense to make `big_content` a callable, instead of `generator`.

asurinsaka replied:
Thanks, I think you are right, use a callable makes much more sense here. Changed.

asurinsaka committed:
DEV-161 add typing

add typing for some tests functions, not sure about
the type of verions_response

anmaxvl reviewed (approved):
just a minor comment. other than that looks good. thanks for cleanup

asurinsaka committed:
DEV-161 refactor big_content

refactor big_content from generator to function,
which makes more sense.
